core/config.py

```diff
@@ -33,7 +33,8 @@ class NoSuchSetting(Exception):
 
     class InvalidConfiguration(Exception):
         """ Raised when configuration validation fails """
-
+        class InvalidSemanticConfiguration(Exception):
+            """ Raised when configuration validation fails due to syntactic errors """
```

Comment:
This should be a subclass of `InvalidConfiguration`, rather than an internal member (i.e., inherit from `InvalidConfiguration`):

```suggestion
    class InvalidSemanticConfiguration(InvalidConfiguration):
        """ Raised when configuration validation fails due to semantic errors """
```